File: modules/wem-admin-ui/src/main/resources/web/admin/common/js/content/ContentCreatedEvent.ts
module api.content {

    export class ContentCreatedEvent extends api.event.Event {

        private contentId: api.content.ContentId;

        constructor(contentId: api.content.ContentId) {
            super();
            this.contentId = contentId;
        }

        public getContentId(): api.content.ContentId {
            return this.contentId;
        }

        static on(handler: (event: ContentCreatedEvent) => void) {
            api.event.Event.bind(api.ClassHelper.getFullName(this), handler);
        }

        static un(handler?: (event: ContentCreatedEvent) => void) {
            api.event.Event.unbind(api.ClassHelper.getFullName(this), handler);
        }

        static fromJson(json: ContentUpdatedEventJson): ContentCreatedEvent {
            return new ContentCreatedEvent(new ContentId(json.contentId));
        }
    }

}

File: modules/core-api/src/main/java/com/enonic/wem/api/content/ContentCreatedEvent.java
package com.enonic.wem.api.content;

import com.google.common.base.MoreObjects;

import com.enonic.wem.api.event.Event;

public final class ContentCreatedEvent
    implements Event
{
    private final ContentId contentId;

    public ContentCreatedEvent( final ContentId contentId )
    {
        this.contentId = contentId;
    }

    public ContentId getContentId()
    {
        return contentId;
    }

    @Override
    public String toString()
    {
        return MoreObjects.toStringHelper( this ).
            add( "contentId", this.contentId ).
            omitNullValues().
            toString();
    }

}
